### PR TITLE
SRV-93 Choose Events for Payment

### PR DESCRIPTION
**Pull request by teghnet**

### CODE REVIEW

File: database/migrations/2018_04_06_103920_create_payments_table.php [hunk 1]
```diff
@@ -15,41 +15,49 @@
...
+                    [
+                        Payment::STATE_NEW,
+                        Payment::STATE_SENT,
+                        Payment::STATE_SUCCESSFUL,
+                        Payment::STATE_FAILED,
+                    ]);
 
-            $table->boolean('completed');
-        });
+                $table->boolean('completed');
```
Comment (PawelPodkalicki) [inline database/migrations/2018_04_06_103920_create_payments_table.php:59]: What is the reason to have `completed` while `state` is present?

File: docker/cron/adpay-payment.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+set -e
+
+if [ ! -e adpay_export_in_progress ]; then
+    touch adpay_export_in_progress
+
+    ./artisan ops:adpay:campaign:export
+    ./artisan ops:adpay:event:export
```
Comment (PawelPodkalicki) [inline docker/cron/adpay-payment.sh]: What is the difference between `adpay-payment.sh` and `adpay-export.sh`?

File: app/Models/Payment.php [hunk 3]
```diff
@@ -88,4 +97,9 @@ public static function fetchPayment(string $transactionId, string $accountAddres
             ->where('account_address', hex2bin($accountAddress))
             ->first();
     }
+
+    public function events(): \Illuminate\Database\Eloquent\Relations\HasMany
```
Comment (PawelPodkalicki) [inline app/Models/Payment.php]: Full path in return type

File: app/Models/EventLog.php [hunk 4]
```diff
@@ -98,6 +99,26 @@ class EventLog extends Model
...
+        return $query->get();
+    }
+
+    public static function fetchEvents(int $paymentId): Collection
+    {
+        return self::where('payment_id', $paymentId)
+            ->get();
+    }
+
+    public function payment(): \Illuminate\Database\Eloquent\Relations\BelongsTo
```
Comment (PawelPodkalicki) [inline app/Models/EventLog.php]: Full path in return type

File: app/Models/EventLog.php [hunk 5]
```diff
@@ -98,6 +99,26 @@ class EventLog extends Model
         'their_userdata' => 'JsonValue',
     ];
 
+    public static function fetchUnpaidEvents(): \Illuminate\Support\Collection
```
Comment (PawelPodkalicki) [inline app/Models/EventLog.php]: Return type could be `Collection`, because it is Eloquent

**Review by PawelPodkalicki — CHANGES_REQUESTED** [review]

File: docker/cron/adpay-payment.sh (hunk 2 above)
Comment (teghnet) [reply docker/cron/adpay-payment.sh]: Right. Fixed.

File: app/Models/EventLog.php (hunk 5 above)
Comment (teghnet) [reply app/Models/EventLog.php]: +1

File: app/Models/EventLog.php (hunk 4 above)
Comment (teghnet) [reply app/Models/EventLog.php]: :ok:

File: database/migrations/2018_04_06_103920_create_payments_table.php (hunk 1 above)
Comment (teghnet) [reply database/migrations/2018_04_06_103920_create_payments_table.php:59]: Backward Compatibility

**Review by PawelPodkalicki — APPROVED** [review]

**Pull request merged**